cmd/server.go

```diff
@@ -108,32 +108,63 @@ func runOnServer(apiOrigin string) {
 	// TODO: get server credentials
 	credentials := client.AgentCredentials{}
 
-	// start ping server to send pings and update agent config
 	beat := client.ServerHeartbeat{CloudID: getCloudID()}
+
+	wsm := WebSocketManager{
+		ConfigChannel:    make(chan client.AgentConfig, 100),
+		HeartbeatChannel: make(chan interface{}, 100),
```

Comment:
My only thought here is possibly moving this channel creation outside of the WebSocketManager. That way devices and servers can use their respective heartbeat types and we don't need to fallback to `interface{}`.

Reply:
if we want to make channels more specific to a type, I'd rather have two channels inside of a manager since these channels otherwise should be passed by pointers to two functions. What do you think?

Reply:
I don't mind it being an `interface{}` simply because the only thing that `WebSocketManager` does with channel messages is `json.Marshal(beat)`. I'd feel differently if it required more type specificity.